### PR TITLE
Opentelemetry metrics backend

### PR DESCRIPTION
**Pull request by Pask423**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -891,26 +890,27 @@ lazy val prometheusBackend = (projectMatrix in file("metrics/prometheus-backend"
   .jvmPlatform(scalaVersions = scala2 ++ scala3)
   .dependsOn(core)
 
-lazy val openTracingBackend = (projectMatrix in file("metrics/open-tracing-backend"))
+lazy val openTelemetryTracingBackend = (projectMatrix in file("metrics/open-telemetry-tracing-backend"))
```
Comment (adamw) [inline build.sbt:893]: if the dependencies are the same I think it can be a single project

File: docs/backends/wrappers/opentelemetrymetrics.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Opentelemetry metrics backend
```
Comment (adamw) [inline docs/backends/wrappers/opentelemetrymetrics.md:1]: same here, I would just have a single "OpenTelemetry" page

File: metrics/open-telemetry/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryMetricsBackend.scala [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  val DefaultFailureCounterName = "sttp3_requests_failure_count"
+  val DefaultRequestHistogramName = "sttp3_request_size_bytes"
+  val DefaultResponseHistogramName = "sttp3_response_size_bytes"
+
+  def apply[F[_], P](
+      delegate: SttpBackend[F, P],
+      openTelemetry: OpenTelemetry,
+      meterConfig: Option[MeterConfig] = None,
+      requestToInProgressCounterNameMapper: Request[_, _] => Option[CollectorConfig] = (_: Request[_, _]) =>
+        Some(CollectorConfig(DefaultRequestsInProgressCounterName)),
```
Comment (adamw) [inline metrics/open-telemetry/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryMetricsBackend.scala:152]: shouldn't we provide some units as default?

**Review by adamw — APPROVED** [review]

File: metrics/open-telemetry/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryMetricsBackend.scala (hunk 3 above)
Comment (Pask423) [reply metrics/open-telemetry/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryMetricsBackend.scala:152]: In fact unit itself is optional and we are using it in many contexts so I am not sure what exactly should be used as default unit

**Pull request merged**